From f7b8cd986e52fffb0880cb86c8681f395dce69d3 Mon Sep 17 00:00:00 2001
From: Andy Ross <andrew.j.ross@intel.com>
Date: Mon, 22 Sep 2014 13:27:25 -0700
Subject: [PATCH 004/117] Camera drivers build integration

Makefile integration for camera ISP & sensor drivers that live under
drivers/external_drivers

Signed-off-by: Andy Ross <andrew.j.ross@intel.com>
---
 drivers/Kconfig                                    |  1 +
 drivers/Makefile                                   |  1 +
 drivers/external_drivers/Kconfig                   | 24 ++++++++++++++++++++++
 drivers/external_drivers/Makefile                  |  1 +
 drivers/external_drivers/camera/Kconfig            |  3 +++
 drivers/external_drivers/camera/Makefile           |  1 +
 drivers/external_drivers/camera/drivers/Kconfig    |  1 +
 drivers/external_drivers/camera/drivers/Makefile   |  1 +
 .../external_drivers/camera/drivers/media/Kconfig  |  5 +++++
 .../external_drivers/camera/drivers/media/Makefile |  7 +++++++
 10 files changed, 45 insertions(+)
 create mode 100644 drivers/external_drivers/Kconfig
 create mode 100644 drivers/external_drivers/Makefile
 create mode 100644 drivers/external_drivers/camera/Kconfig
 create mode 100644 drivers/external_drivers/camera/Makefile
 create mode 100644 drivers/external_drivers/camera/drivers/Kconfig
 create mode 100644 drivers/external_drivers/camera/drivers/Makefile
 create mode 100644 drivers/external_drivers/camera/drivers/media/Kconfig
 create mode 100644 drivers/external_drivers/camera/drivers/media/Makefile

diff --git a/drivers/Kconfig b/drivers/Kconfig
index 5e0eb15..577e818 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -173,3 +173,4 @@ source "drivers/phy/Kconfig"
 source "drivers/powercap/Kconfig"
 
 endmenu
+source "drivers/external_drivers/Kconfig"
diff --git a/drivers/Makefile b/drivers/Makefile
index cde0860..9c51bba 100644
--- a/drivers/Makefile
+++ b/drivers/Makefile
@@ -156,3 +156,4 @@ obj-$(CONFIG_IPACK_BUS)		+= ipack/
 obj-$(CONFIG_NTB)		+= ntb/
 obj-$(CONFIG_FMC)		+= fmc/
 obj-$(CONFIG_POWERCAP)		+= powercap/
+obj-$(CONFIG_EXTERNAL_DRIVERS)+= external_drivers/
diff --git a/drivers/external_drivers/Kconfig b/drivers/external_drivers/Kconfig
new file mode 100644
index 0000000..f509839
--- /dev/null
+++ b/drivers/external_drivers/Kconfig
@@ -0,0 +1,24 @@
+menuconfig EXTERNAL_DRIVERS
+	bool "Enable support to external drivers"
+	depends on X86
+	default y
+	help
+	  External drivers are drivers located outside this kernel tree
+	  but allowed to be configured as builtin.
+	  External drivers directories will behave as a regular directory
+	  inside kernel tree.
+
+if EXTERNAL_DRIVERS
+
+menuconfig EXTERNAL_CAMERA
+        bool "Enable support to Intel camera drivers"
+        depends on EXTERNAL_DRIVERS
+        default y
+        help
+          Enable support to external Intel camera drivers.
+
+if EXTERNAL_CAMERA
+source "drivers/external_drivers/camera/Kconfig"
+endif
+
+endif
diff --git a/drivers/external_drivers/Makefile b/drivers/external_drivers/Makefile
new file mode 100644
index 0000000..ec8bcdf
--- /dev/null
+++ b/drivers/external_drivers/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_EXTERNAL_CAMERA)		+= camera/
diff --git a/drivers/external_drivers/camera/Kconfig b/drivers/external_drivers/camera/Kconfig
new file mode 100644
index 0000000..e51f2f1
--- /dev/null
+++ b/drivers/external_drivers/camera/Kconfig
@@ -0,0 +1,3 @@
+if EXTERNAL_CAMERA
+source drivers/external_drivers/camera/drivers/Kconfig
+endif
diff --git a/drivers/external_drivers/camera/Makefile b/drivers/external_drivers/camera/Makefile
new file mode 100644
index 0000000..a175427
--- /dev/null
+++ b/drivers/external_drivers/camera/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_EXTERNAL_CAMERA) += drivers/
diff --git a/drivers/external_drivers/camera/drivers/Kconfig b/drivers/external_drivers/camera/drivers/Kconfig
new file mode 100644
index 0000000..3a4f2c3
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/Kconfig
@@ -0,0 +1 @@
+source drivers/external_drivers/camera/drivers/media/Kconfig
diff --git a/drivers/external_drivers/camera/drivers/Makefile b/drivers/external_drivers/camera/drivers/Makefile
new file mode 100644
index 0000000..f00d749
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/Makefile
@@ -0,0 +1 @@
+obj-y += media/
diff --git a/drivers/external_drivers/camera/drivers/media/Kconfig b/drivers/external_drivers/camera/drivers/media/Kconfig
new file mode 100644
index 0000000..f555355
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/media/Kconfig
@@ -0,0 +1,5 @@
+#
+# Kconfig for Camera drivers
+#
+source drivers/external_drivers/camera/drivers/media/pci/Kconfig
+source drivers/external_drivers/camera/drivers/media/i2c/Kconfig
diff --git a/drivers/external_drivers/camera/drivers/media/Makefile b/drivers/external_drivers/camera/drivers/media/Makefile
new file mode 100644
index 0000000..37171c4
--- /dev/null
+++ b/drivers/external_drivers/camera/drivers/media/Makefile
@@ -0,0 +1,7 @@
+#
+# Makefile for camera drivers.
+#
+
+obj-y += pci/
+obj-y += i2c/
+
-- 
1.9.3

